core/src/main/scala/com/rackspace/com/papi/components/checker/util/XPathExpressionPool.scala

```diff
@@ -104,9 +104,10 @@ private class XPathExpressionFactory(private val expression : String, private va
 }
 
 
-private class XPath2ExpressionFactory(private val expression : String, private val nc : NamespaceContext) extends XPathExpressionFactory(expression, nc) {
+private class XPathNExpressionFactory(private val expression : String, private val nc : NamespaceContext, private val version : Int) extends XPathExpressionFactory(expression, nc) {
   override def makeObject = {
     val xpath = (new net.sf.saxon.xpath.XPathFactoryImpl()).newXPath()
```

Comment:
I get that, I was just saying we could consolidate all the `version match` blocks into the `private` factory implementation itself so that it was streamlined everywhere else.